### ISSUE TITLE
Integrate Netlify for Deploy Previews

### ISSUE THREAD

**Issue by praveenscience** [praveenscience/Internship-LMS-FrontEnd#20]
As we move up to the coding phase now, let's start implementing the Deploy Previews feature of Netlify. When someone creates a PR, the netlify service should send a link to show how it looks.

You can find more information here: [Introducing Deploy Previews in Netlify](https://www.netlify.com/blog/2016/07/20/introducing-deploy-previews-in-netlify/). This is up for grabs people. Anyone from DWoC or SWoC can pick it up.

When you pick up, please answer the following in bullet points as answers:

- Do you understand what is required? Do you need any help with the issue?
- How much Netlify integration knowledge you have? You wanna try or you know it and you wanna work?
- Which programme you're working on? DWoC or SWoC or both?

As always, reach out to me in case of anything. 😁

**Comment by avinashupadhya99** [comment]
Hi, can I work on this?

> Do you understand what is required? Do you need any help with the issue?

Yes, it is essentialy setting up a CICD for the application


> How much Netlify integration knowledge you have? You wanna try or you know it and you wanna work?

I haven't done much with Netlify but I have worked with other hosting services. I want to learn doing it on Netlify and will do so.


> Which programme you're working on? DWoC or SWoC or both?

I am working on DWoC


Can you please assign this to me?

**Comment by praveenscience** [comment]
Hi @avinashupadhya99, yes, technically it's setting up CI/CD for only the PRs... 😁

> Can you please assign this to me?

Cool, you can start working on this. 😁 Keep me updated with the progress.

**Comment by avinashupadhya99** [comment]
Hey, I looked up deploy previews on Netlify and tested it out with my repo. There are no code changes necessary for deploy previews. Once the repository is set up for deployment on Netlify, any PR made to the repository will have a deploy preview. There are no changes required.

Please let me know if you plan on implementing CICD for the entire application. I can work on that. 🥂

**Comment by praveenscience** [comment]
Yes, @avinashupadhya99. What needs to be done? Should we do it together?

**Comment by praveenscience** [comment]
@avinashupadhya99 I'll set this thing up. Once I have done it, I'll ask you to update some code changes so that you get the points for looking into. Hope that helps?

**Comment by avinashupadhya99** [comment]
> You only need to add commit hooks to your Netlify sites, we take care of the rest.

Yes you just need to add the repository on Netlify. Everything else is taken care by them.

> @avinashupadhya99 I'll set this thing up. Once I have done it, I'll ask you to update some code changes so that you get the points for looking into. Hope that helps?

No issue. I will work on other *issues* that come up

**Comment by praveenscience** [comment]
@avinashupadhya99 This one is still your responsibility. Thanks for getting me a clarity on this. 😁 Will award the right points for you.

**Comment by avinashupadhya99** [comment]
I'm sorry, I didn't get it. Should I set up the netlify on my account? If yes, I don't think that is possible because I would need access to the repository.

If it is regarding the points for DWoC, I'm fine with it. 😄 I got to learn something new. 😁

**Comment by praveenscience** [comment]
@avinashupadhya99 Send a random PR to fix something. I'll get you the points you deserve. Can you please do it now?

**Comment by Uyadav207** [comment]
@praveenscience and @avinashupadhya99  , i have worked with netlify CI, i might help you both. infact my all website are netlify deployed.

**Comment by praveenscience** [comment]
@Uyadav207 Sure, let's jump on a call.

**Comment by satyampandeygit** [comment]
Hello Praveen I want to work on this issue, please find below answers to the above questions

- I understand what is required if any help will be required I will ask here itself
- I know how Netlify works but I have not done such integration before but would love to do this one
- I am in GSSOC, if you will change tags then it will add to my points. I am fine in another way also would love to contribute anyways

**Comment by praveenscience** [comment]
@satyampandeygit Did you message something here?

**Comment by ampsteric** [comment]
@praveenscience as it is the admin who needs to link his repo to his own netlify credentials, should I close this issue?

**Comment by praveenscience** [comment]
@ampsteric Once I do it, let's close it. I'll self-assign it to myself.

**Comment by Abhishekkr93** [comment]
can I do this, could you please assign me.

**Comment by subhangi2731** [comment]
please assign this to me

**Comment by praveenscience** [comment]
@subhangi2731 I have assigned it to you.

**Comment by gargijs** [comment]
I want to work on this

**Comment by Abhishekkr93** [comment]
I want to fix this, please assign me.

**Comment by praveenscience** [comment]
Let's meet on Monday to check the status of this:
- @subhangi2731 (It's been 12 days and no response)
- @gargijs (You have time till Monday!)
- @Abhishekkr93 (You have time till Monday!)

**Comment by subhangi2731** [comment]
@praveenscience I will make a pr by today

**Comment by subhangi2731** [comment]
Sorry I had exams in my college @praveenscience

**Comment by praveenscience** [comment]
Okay thanks, let me know the progress folks! 🔥

- @subhangi2731 (any updates?)
- @gargijs (any updates?)
- @Abhishekkr93 (any updates?)

**Comment by praveenscience** [comment]
Any updates folks? By tomorrow, if there's no activity, I am happy to do it myself and implement this... 🔥

**Comment by vave4** [comment]
Can I try solving this issue?

**Comment by praveenscience** [comment]
Yes, sure @vave4

**Comment by praveenscience** [comment]
Any updates @vave4?
